### PR TITLE
fix(ci): whitenoise lint failed when pull request

### PR DESCRIPTION
**Pull request by zaunist**
**What type of PR is this?**

Fix whitenoise lint error

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes [523](https://github.com/envoyproxy/gateway/issues/523)

### CODE REVIEW

**Comment by Xunzhuo** [comment]
image

Thanks for working on it, I did not look deep into this issue, but it seems like still throw an error in Pull Request.

**Comment by zaunist** [comment]
> image
> Thanks for working on it, I did not look deep into this issue, but it seems like still throw an error in Pull Request.

You're right, I'm not entirely sure about the exact problem either. This is my first test, and I will further study how to solve this issue carefully.

**Comment by zaunist** [comment]
Please temporarily ignore the previous commit records above. I am currently investigating the cause of the issue on my own branch.

**Comment by zaunist** [comment]
Hi, @Xunzhuo. I've identified the issue. In the github pull request CI environment, there is no branch information available, causing the git diff command to fail. Therefore, I added a git fetch command to retrieve the remote branch information and prefixed the branch names with "origin" to match them.

Here is my debug info for reference.

image

https://github.com/zaunist/gateway/actions/runs/6262614026/job/17005194717

**Comment by arkodg** [comment]
@zaunist DCO is failing, can you squash, sign and push again ?

**Comment by zaunist** [comment]
> @zaunist DCO is failing, can you squash, sign and push again ?

Done

**Comment by arkodg** [comment]
DCO is still failing, you might need to squash commits and then resign and push

**Review by zirain — APPROVED** [review]

File: tools/src/whitenoise.sh [hunk 1]
```diff
@@ -1,7 +1,8 @@
 #!/bin/bash
 TRAILING_WHITESPACE=0
 # only check changed docs in pr
-for file in $(git diff --cached --name-only --diff-filter=ACRMTU $GITHUB_BASE_REF | grep "\.md"); do
+git fetch
+for file in $(git diff --cached --name-only --diff-filter=ACRMTU origin/$GITHUB_BASE_REF| grep "\.md"); do
```
Comment (arkodg) [inline tools/src/whitenoise.sh:5]: thinking out loud, why isn't `git diff --name-only` good enough here ?

Comment (zaunist) [reply tools/src/whitenoise.sh:5]: The issue here is that the GitHub Actions environment does not have the branch information for Git, which causes an error when using `git diff` to compare the differences. To resolve this, I simply added a `git fetch` command to retrieve the branch information from the remote repository. For further debugging information, you can refer to this link: [https://github.com/zaunist/gateway/actions/runs/6262614026/job/17005194717](https://github.com/zaunist/gateway/actions/runs/6262614026/job/17005194717).

Comment (zaunist) [reply tools/src/whitenoise.sh:5]: @arkodg If we don't provide branch information, how can we compare the differences between the fork and main branches? I'm not quite understanding this point. Could you provide an example, please?

**Review by Xunzhuo — APPROVED** [review]
I am OK with current approach, thanks for working on it @zaunist.

Actually I think this one is not that high priority, if you would like to help, we can try to pick some important issues for you : )

**Comment by zaunist** [comment]
> I am OK with current approach, thanks for working on it @zaunist.
> 
> Actually I think this one is not that high priority, if you would like to help, we can try to pick some important issues for you : )

Willing to make more contributions

**Pull request merged**